zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/adapters/ClientAdapter.java

```diff
@@ -8,20 +8,25 @@
 package io.camunda.exporter.adapters;
 
 import io.camunda.exporter.ExporterResourceProvider;
+import io.camunda.exporter.config.ConnectionTypes;
 import io.camunda.exporter.config.ExporterConfiguration;
 import io.camunda.exporter.schema.SchemaManager;
 import io.camunda.exporter.schema.SearchEngineClient;
 import io.camunda.exporter.store.BatchRequest;
-import io.camunda.search.connect.configuration.ConnectConfiguration;
 import java.io.IOException;
 
 public interface ClientAdapter {
-  void createClient(final ConnectConfiguration config);
 
-  SearchEngineClient createSearchEngineClient();
+  static ClientAdapter of(final ExporterConfiguration configuration) {
+    return switch (ConnectionTypes.from(configuration.getConnect().getType())) {
```

Comment:
We can have a default: throw which would validate the connection type, the difference being it would throw in open (which would retry, and the exporter would be stuck in a retry loop), verses throwing in configure (which I think just crashes and doesn't retry? Correct me if I'm wrong).

Reply:
I think configuration validation should be done in `config` anyway, we have a whole ticket to start introducing it.

Also, there won't be a case for default, since converting to the enum type will fail in case of a wrong value. (even if I removed my throws exception statement there, it will throw `IllegalArgumentException`)

Reply:
I was just thinking of having `configuration.getConnect().getType()` in the switch statement to catch it but if we're doing validation in `configure` then keep it that way to be consitent.